frontend/tests/controllers/ContestUsersTest.php

```diff
@@ -68,4 +68,75 @@ public function testContestActivityReport() {
         $this->assertEquals(0, $response['events'][0]['ip']);
         $this->assertEquals('open', $response['events'][0]['event']['name']);
     }
+
+    public function testContestParticipantsReport() {
+        // Get a contest
+        $contestData = ContestsFactory::createContest(new ContestParams(['requests_user_information' => 'optional']));
+
+        for ($i = 0; $i < 3; $i++) {
+            // Create users
+            $user[$i] = UserFactory::createUser();
+
+            // Add users to our private contest
+            ContestsFactory::addUser($contestData, $user[$i]);
+        }
+
+        $userLogin = self::login($user[0]);
+
+        $showContestIntro = ContestController::showContestIntro(new Request([
+            'auth_token' => $userLogin->auth_token,
+            'contest_alias' => $contestData['request']['alias'],
+        ]));
+
+        // Explicitly join contest
+        ContestController::apiOpen(new Request([
+            'contest_alias' => $contestData['request']['alias'],
+            'auth_token' => $userLogin->auth_token,
+            'privacy_git_object_id' => $showContestIntro['git_object_id'],
+            'statement_type' => $showContestIntro['statement_type'],
+            'share_user_information' => 1,
+        ]));
+
+        // Call API
+        $directorLogin = self::login($contestData['director']);
+
+        $r = new Request([
+            'contest_alias' => $contestData['request']['alias'],
+            'auth_token' => $directorLogin->auth_token
+        ]);
+
+        $response = ContestController::apiContestants($r);
+
+        $this->assertEquals(1, count($response['contestants']));
```

Comment:
```suggestion
        $this->assertEquals(3, count($response['contestants']));
```

pero luego hay que checar que la info de los que no dieron permiso está como `null`.

Reply:
listo